scheduler/core/calculations/targetinfo.py

```diff
@@ -64,6 +65,64 @@ class TargetInfo:
     def mean_airmass(self, interval: npt.NDArray[int]):
         return np.mean(self.airmass[interval])
 
+    @staticmethod
+    def from_dict(ti_dict: Dict) -> 'TargetInfo':
+        return TargetInfo(coord=SkyCoord(ra=np.array(ti_dict['coord']['ra']) * u.deg,
+                                         dec=np.array(ti_dict['coord']['dec']) * u.deg,
+                                         frame=ti_dict['coord']['frame']),
+                          alt=Angle(np.array(ti_dict['alt']['value']), unit=ti_dict['alt']['unit']),
+                          az=Angle(np.array(ti_dict['az']['value']), unit=ti_dict['az']['unit']),
+                          par_ang=Angle(np.array(ti_dict['par_ang']['value']), unit=ti_dict['par_ang']['unit']),
+                          hourangle=Angle(np.array(ti_dict['hourangle']['value']), unit=ti_dict['hourangle']['unit']),
+                          airmass=np.array(ti_dict['airmass']),
+                          sky_brightness=np.array(ti_dict['sky_brightness']),
+                          visibility_slot_idx=np.array(ti_dict['visibility_slot_idx']),
+                          visibility_slot_filter=np.array(ti_dict['visibility_slot_filter']),
```

Comment:
For these two, we might not need them both: I think if we use the `NightEvents`, we can just figure out one from the other? If it's not going to take up too much space, though, we may as well include them both in the cache.